### PR TITLE
Rename MAIN & REGTEST to PROD & DEV 

### PR DESCRIPTION
**Pull request by Naviabheeman**
Since Tapyrus has only two modes they are being renamed from MAIN and REGTEST to PROD and DEV. 
Class CChainParamsBase is renamed to CFederationParams
Classes CMainParams and CRegtestParams are renamed to CProductionParams and CDevelopmentParams.

These changes are made over #37 code base. This PR can be merged after #37

### CODE REVIEW

File: src/test/chainparamsbase_tests.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@ extern void noui_connect();
 
 struct ChainParamsBaseTestingSetup {
```
Comment (azuchi) [inline src/test/chainparamsbase_tests.cpp]: You should also rename to FederationParamsBaseTestingSetup. filename and following code too.

Comment (Naviabheeman) [reply src/test/chainparamsbase_tests.cpp]: yes! will correct it.

File: src/util.h [hunk 2]
```diff
@@ -261,8 +261,8 @@ class ArgsManager
     void ForceSetArg(const std::string& strArg, const std::string& strValue);
 
     /**
-     * Looks for -regtest and returns the appropriate BIP70 chain name.
-     * @return CBaseChainParams::MAIN by default; raises runtime error if an invalid combination is given.
+     * Looks for -dev and returns the appropriate BIP70 chain name.
+     * @return CFederationParams::MAIN by default; raises runtime error if an invalid combination is given.
```
Comment (azuchi) [inline src/util.h]: Is'nt it `TAPYRUS_OP_MODE::PROD`?

**Comment by azuchi** [comment]
In `doc/tapyrus` still remain main, regtest.

**Pull request merged**